### ISSUE TITLE
Missing "./build/esm-debug" specifier in "socket.io-client" package 

### ISSUE THREAD

**Issue by wvhulle** [socketio/socket.io#4635]
**Describe the bug**

The documentation is unable to help me to set up debugging in the client when installed through npm.
It only refers to cdn, but i cannot find anywhere on the Socket IO website or google how to import the esm-debug version. 

Importing `socket.io-client/build/esm-debug` in Vite results in 
```
Internal server error: Missing "./build/esm-debug" specifier in "socket.io-client" package
```

**To Reproduce**

localStorage.debug = "*"
No logs shown.

*Server*

socket.io 4.6.1

*Client*

socket.io-client 4.6.1

**Expected behavior**

Show debug statements in browser

Not possible

**Platform:**
Edge and Firefox
Vite

**Additional context**

**Comment by darrachequesne** [comment]
Hi! We could indeed export the debug build, which is currently only available for the Node.js platform ([here](https://github.com/socketio/socket.io-client/blob/89175d0481fc7633c12bb5b233dc3421f87860ef/package.json#L26)).

That being said, I am not sure whether Vite supports the `exports` field, something like:

```js
{
  // ...
  "exports": {
    "/debug": {
      "import": "./build/esm-debug/index.js",
      "require": "./build/cjs/index.js",
      "types": "./build/esm/index.d.ts"
    }
  },
}
```

And then: `import { io } from "socket.io-client/debug"`

Reference: https://nodejs.org/api/packages.html#package-entry-points

What do you think?

**Comment by wvhulle** [comment]
I would have to check if vite supports that specifically for the client but I have found another way. In vite it is also possible to declare the path to the debug version as resolve alias. Then you can import it with a relative path in the client code, at least on the dev server. There is also some other issue with XMLrequest that can be resolved like this. 

Besides that, in the server I also found out that it is possible to toggle debugging on and off by importing debug from debug and calling debug.enable. I prefer this to yet another environment variable.

Maybe this could be documented?

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-client/commit/781d753a626d01e675056a2ff4e27f5dd599564f, included in version [4.7.0](https://github.com/socketio/socket.io-client/releases/tag/4.7.0).

Please reopen if needed.
